src/main/java/org/opentripplanner/updater/trip/TimetableSnapshotManager.java

```diff
@@ -99,19 +79,11 @@ public TimetableSnapshotManager(
    * to the snapshot to release resources.
    */
   public TimetableSnapshot getTimetableSnapshot() {
-    // Try to get a lock on the buffer
-    if (bufferLock.tryLock()) {
-      // Make a new snapshot if necessary
-      try {
-        commitTimetableSnapshot(false);
-        return snapshot;
-      } finally {
-        bufferLock.unlock();
-      }
-    }
-    // No lock could be obtained because there is either a snapshot commit busy or updates
-    // are applied at this moment, just return the current snapshot
-    return snapshot;
+    return snapshot.get();
+  }
+
+  public TimetableSnapshot getTimetableSnapshotBuffer() {
```

Comment:
Can you add Javadoc under what circumstances you want to have the buffer rather than the snapshot?

Reply:
Done